drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -91,6 +107,27 @@ local function temperature_attr_handler(driver, device, ib, response)
   device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperature({value = temp, unit = unit}))
 end
 
+local temp_attr_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value ~= nil then
+      local temp = ib.data.value / 100.0
+      local unit = "C"
+      device:set_field(TEMP_BOUND_RECEIVED..minOrMax, temp)
+      local min = device:get_field(TEMP_BOUND_RECEIVED..TEMP_MIN)
+      local max = device:get_field(TEMP_BOUND_RECEIVED..TEMP_MAX)
+      if min ~= nil and max ~= nil then
+        if min < max then
+          device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperatureRange({ value = { minimum = min, maximum = max }, unit = unit }))
```

Comment:
There doesn't seem to be any mechanism preventing multiple endpoints from interfering with one another. Perhaps the fields need to be scoped to an endpoint?